plugins/parsers/avro/parser_test.go

```diff
@@ -0,0 +1,86 @@
+package avro
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/inputs/file"
+	"github.com/influxdata/telegraf/plugins/parsers/influx"
+	"github.com/influxdata/telegraf/testutil"
+)
+
+func TestCases(t *testing.T) {
+	// Get all directories in testdata
+	folders, err := os.ReadDir("testdata")
+	require.NoError(t, err)
+	// Make sure testdata contains data
+	require.NotEmpty(t, folders)
+
+	testdataParser := &influx.Parser{}
+	require.NoError(t, testdataParser.Init())
+
+	for _, f := range folders {
+		fname := f.Name()
+		testdataPath := filepath.Join("testdata", fname)
+		configFilename := filepath.Join(testdataPath, "telegraf.conf")
+		expectedFilename := filepath.Join(testdataPath, "expected.out")
+		expectedErrorFilename := filepath.Join(testdataPath, "expected.err")
+
+		t.Run(fname, func(t *testing.T) {
+			inputs.Add("file", func() telegraf.Input {
+				return &file.File{}
+			})
+
+			var expected []telegraf.Metric
+			if _, err := os.Stat(expectedFilename); err == nil {
+				var err error
+				expected, err = testutil.ParseMetricsFromFile(expectedFilename, testdataParser)
+				require.NoError(t, err)
+			}
+
+			// Read the expected errors if any
+			var expectedErrors []string
+
+			if _, err := os.Stat(expectedErrorFilename); err == nil {
+				var err error
+				expectedErrors, err = testutil.ParseLinesFromFile(expectedErrorFilename)
+				require.NoError(t, err)
+				require.NotEmpty(t, expectedErrors)
+			}
+
+			// Set up error catching
+			var acc testutil.Accumulator
+			var actualErrors []string
+			var actual []telegraf.Metric
+
+			// Configure the plugin
+			cfg := config.NewConfig()
+			err := cfg.LoadConfig(configFilename)
+			if err != nil {
+				actualErrors = append(actualErrors, err.Error())
+			} else {
+				for _, input := range cfg.Inputs {
+					err := input.Init()
+					if err != nil {
+						actualErrors = append(actualErrors, err.Error())
+					} else {
+						err := input.Gather(&acc)
+						if err != nil {
+							actualErrors = append(actualErrors, err.Error())
+						}
+					}
+				}
+			}
```

Comment:
I think we don't need to continue if loading the config fails or if initializing the parser fails. Test-cases should be for runtime errors only...
```suggestion
			require.NoError(t, err)

			for _, input := range cfg.Inputs {
				require.NoError(t, input.Init())

				if err := input.Gather(&acc); err != nil {
					actualErrors = append(actualErrors, err.Error())
				}
			}
```